libevmasm/AssemblyItem.cpp

```diff
@@ -26,27 +26,35 @@ using namespace std;
 using namespace dev;
 using namespace dev::eth;
 
+static_assert(sizeof(size_t) <= 8, "size_t must be at most 64-bits wide");
+
 AssemblyItem AssemblyItem::toSubAssemblyTag(size_t _subId) const
 {
 	assertThrow(data() < (u256(1) << 64), Exception, "Tag already has subassembly set.");
-
 	assertThrow(m_type == PushTag || m_type == Tag, Exception, "");
+	size_t tag = size_t(u256(data()) & 0xffffffffffffffffL);
```

Comment:
Hm, I'm still not sure about the `L` suffix. Perhaps it would be best to use `u256(data()) & u256("0xffffffffffffffff")`?

Reply:
`L` suffix should work in specifying 64 bit literals, but can do `u256` too.

Reply:
Sorry, it should be `LL` for long long (>= 64bit).

Reply:
Or what about `size_t(-1)` - we actually assume size_t to be 64 bits here. That should be a static assert, shouldn't it?

Reply:
Perhaps even `ull`

Reply:
Probably `size_t(-1)` is safest.